docs/deployment/automated-local.rst

```diff
@@ -0,0 +1,227 @@
+
+	If you always connect the drive(s) to the same physical hardware path, e.g. the same
+	eSATA port, then you can make a more precise udev rule.
+
+	Execute ``udevadm monitor`` and connect a drive to the port you intend to use.
+	You should see a flurry of events, find those regarding the `block` subsystem.
+	Pick the event whose device path ends in something similar to a device file name,
+	typically`sdX/sdXY`. Use the event's device path and replace `sdX/sdXY` after the
+	`/block/` part in the path with a star (\*). For example:
+	`DEVPATH=="/devices/pci0000:00/0000:00:11.0/ata3/host2/target2:0:0/2:0:0:0/block/*"`.
+
+	Reboot a few times to ensure that the hardware path does not change: on some motherboards
+	components of it can be random. In these cases you cannot use a more accurate rule,
+	or need to insert additional stars for matching the path.
+
+The "systemd" tag in conjunction with the SYSTEMD_WANTS environment variable has systemd
+launch the "automatic-backup" service, which we will create next, as the
+``/etc/backups/automatic-backup.service`` file:
+
+.. code-block:: ini
+
+    [Service]
+    Type=oneshot
+    ExecStart=/etc/backups/run.sh
+
+Now, create the main backup script, ``/etc/backups/run.sh``. Below is a template,
+modify it to suit your needs (e.g. more backup sets, dumping databases etc.).
+
+.. code-block:: bash
+
+	#!/bin/bash -ue
+	which python borg > /dev/null
+
+	# The udev rule is not terribly accurate and may trigger our service before
+	# the kernel has finished probing partitions. Sleep for a bit to ensure
+	# the kernel is done.
+	#
+	# This can be avoided by using a more precise udev rule, e.g. matching
+	# a specific hardware path and partition.
+	sleep 5
+
+	#
+	# Script configuration
+	#
+
+	# The backup partition is mounted there
+	MOUNTPOINT=/mnt/backup
+
+	# This is the location of the Borg repository
+	TARGET=$MOUNTPOINT/borg-backups/backup.borg
+
+	# Archive name schema.
+	# If you intend to use prune and have more than one backup set, perhaps
+	# swap hostname and date to be compliant with prune's --prefix option.
+	DATE=$(date --iso-8601)-$(hostname)
+
+	# This is the file that will later contain UUIDs of registered backup drives
+	DISKS=/etc/backups/backup.disks
+
+	# Find whether the connected block device is a backup drive
+	for uuid in $(lsblk -nlpo uuid)
+	do
+	        if grep -qF $uuid $DISKS; then
+	                break
+	        fi
+	        uuid=
+	done
+
+	if [ ! $uuid ]; then
+	        echo "No backup disk found, exiting"
+	        exit 0
+	fi
+
+	echo "Disk $uuid is a backup disk"
+	disk_path=/dev/disk/by-uuid/$uuid
+	# Mount file system if not already done. This assumes that if something is already
+	# mounted at $MOUNTPOINT, it is the backup drive. It won't find the drive if
+	# it was mounted somewhere else.
+	(mount | grep $MOUNTPOINT) || mount $disk_path $MOUNTPOINT
+	drive=$(lsblk -snlpo name $disk_path | head -n1)
+	echo "Drive path: $drive"
```

Comment:
Oh, GitHub actually understands enough rST to highlight `.. code-block::` correctly. Neat.